include/lbann/layers/regularizers/layer_norm.hpp

```diff
@@ -247,6 +262,26 @@ void layer_norm_layer<TensorDataType, Layout, Device>::setup_data(
   data_type_layer<TensorDataType>::setup_data(max_mini_batch_size);
   const auto& output_dims = this->get_output_dims();
   std::vector<size_t> out_dims{output_dims.begin(), output_dims.end()};
+
+  unsigned int start_dim;
+  if (m_start_dim >= 0) {
+    start_dim = static_cast<unsigned int>(m_start_dim);
+  }
+  else {
+    start_dim = static_cast<unsigned int>(out_dims.size() + m_start_dim);
+  }
+  if (start_dim < 0 || start_dim >= output_dims.size()) {
```

Comment:
The first check (`start_dim < 0`) isn't needed since an `unsigned int` cannot be negative.

Reply:
Thanks! Modified to int because this check is used for verifying bad user inputs, which might be `-100`.